### PR TITLE
Add python libraries sklearn and nltk

### PR DESCRIPTION
**Pull request by chen386**

### CODE REVIEW

**Comment by coatless** [comment]
Need a changelog entry. What class is this for?

**Comment by chen386** [comment]
CS 105. Why is the checking failing?

**Comment by coatless** [comment]
Need to add an entry into the `ChangeLog.md` file describing the change.

Also, do you need it in `plbase` or `grader-python`?

**Comment by mwest1066** [comment]
@chen386 are you going to also preload these in the python trampoline?

**Comment by mwest1066** [comment]
@chen386 I get a docker build failure with `centos7-plbase/Dockerfile`. Have you tested this?

File: lib/python-caller-trampoline.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # Errors are signaled by exiting with non-zero exit code
 # Exceptions are not caught and so will trigger a process exit with non-zero exit code (signaling an error)
 
-import sys, os, json, importlib, copy, base64, io, matplotlib, signal
+import sys, os, json, importlib, copy, base64, io, matplotlib, signal, sklearn, nltk
```
Comment (nwalters512) [inline lib/python-caller-trampoline.py]: I want someone to audit this for impact on trampoline startup time - of course it's free via forking after the initial load, but I'd just rather know what the impact is.

Comment (mwest1066) [reply lib/python-caller-trampoline.py]: I believe that local testing by Binglin showed a +200ms impact on worker startup time.

**Comment by nwalters512** [comment]
The current test failures are caused by how we handle images in CI - #1688 will address that. Once that's on master, we can merge back into this branch and verify that the tests are then green.

**Comment by chen386** [comment]
@nwalters512 Thanks for fixing the issue. What issues have `numpy` caused in the past on random seeding?

**Comment by nwalters512** [comment]
@chen386 every forked worker would end up with the same seed, so we would effectively kill randomization. If you look at https://github.com/PrairieLearn/PrairieLearn/blob/42dde84826d6a8263f36d0be5b28c0d1bef6ffa5/lib/python-caller-trampoline.py#L43, we now re-seed `numpy` and the built-in `random` on every fork so we're guaranteed to have "proper" randomness. I looked around a bit and it seems like `sklearn` uses `numpy` for randomness, so this should hopefully not be an issue.

**Comment by nwalters512** [comment]
@chen386 the other PR got merged, if you can pull in master and push the changes, we should be good to go!

**Comment by nwalters512** [comment]
@chen386 this will be available for use locally once the image builds - keep an eye on #announce on Slack to see when it lands on prod.

**Pull request merged**